features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/PurchaseGiftCardViewModel.kt

```diff
@@ -94,16 +99,48 @@ class PurchaseGiftCardViewModel @Inject constructor(
         return null
     }
 
-    suspend fun getMerchantById(): ResponseResource<GetDataMerchantIdResponse?>? {
-        purchaseGiftCardDataMerchant?.merchantId?.let { id ->
-            repository.getDashDirectEmail()?.let { email ->
-                return repository.getMerchantById(
-                    merchantId = id,
-                    includeLocations = false,
-                    userEmail = email
-                )
-            }
+    suspend fun getMerchantById(merchantId: Long): ResponseResource<GetMerchantByIdResponse?>? {
+        repository.getDashDirectEmail()?.let { email ->
+            return repository.getMerchantById(
+                merchantId = merchantId,
+                includeLocations = false,
+                userEmail = email
+            )
```

Comment:
currently `purchaseGiftCardDataMerchant` is only valid when the user presses `Next`, but we need to query this information at the beginning.

However, if we didn't have a merchant field in the fragment and the view model, but only one in the viewModel -- I could make changes to simplify code related to the selected merchant.